# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/powerpc/mol/vconfig-devfs.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- mol-2003-06-20/src/vconfig/main.c	2003-06-12 13:00:59.000000000 +0200
+++ mol-2003-06-20-devfs/src/vconfig/main.c	2003-06-21 13:37:54.000000000 +0200
@@ -99,10 +99,12 @@
 	struct vt_stat 	vts;
 	int 		fd;
 	char 		vtname[32];
+	char		vtname2[32];
 	struct termios 	tio;	
 
-	if ((fd = open("/dev/tty0", O_WRONLY)) < 0) {
-		fprintf(stderr, "failed to open /dev/tty0: %s\n", strerror(errno));
+	if ( (fd = open("/dev/tty0", O_WRONLY)) < 0 &&
+	     (fd = open("/dev/vc/0", O_WRONLY)) < 0 ) {
+		fprintf(stderr, "failed to open /dev/tts0 or /dev/vc/0: %s\n", strerror(errno));
 		return 1;
 	}
 
@@ -128,10 +130,12 @@
 	/* open our vt */
 	sprintf(vtname,"/dev/tty%d",vt_no);
 	if ((console_fd = open(vtname, O_RDWR )) < 0) {
-		fprintf(stderr, "Cannot open %s (%s)\n",vtname, strerror(errno));
-		return 1;
+		sprintf(vtname2,"/dev/vc/%d",vt_no);
+		if ((console_fd = open(vtname2, O_RDWR )) < 0) {
+			fprintf(stderr, "Cannot open %s or %s (%s)\n",vtname, vtname, strerror(errno));
+			return 1;
+		}
 	}
-
 	printf("Running MOL video configurator on VT%d\n", vt_no );
 #if 0
 	/* set controlling tty */
